src/screens/TopicPageScreen/elements/ButtonFollowing.js

```diff
@@ -1,36 +1,42 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import {View, TouchableNativeFeedback, Text, StyleSheet} from 'react-native';
 import {fonts, normalize, normalizeFontSize} from '../../../utils/fonts';
 import {COLORS} from '../../../utils/theme';
 
-const ButtonFollowing = ({handleSetUnFollow}) => {
+const ButtonFollowing = ({handleSetUnFollow, followType = 'signed'}) => {
   return (
     <TouchableNativeFeedback onPress={handleSetUnFollow}>
-      <View style={styles.buttonFollowing}>
-        <Text style={styles.textButtonFollowing}>Joined</Text>
+      <View style={styles.buttonFollowing(followType)}>
+        <Text style={styles.textButtonFollowing(followType)}>Joined</Text>
       </View>
     </TouchableNativeFeedback>
   );
 };
 
+ButtonFollowing.propTypes = {
+  handleSetUnFollow: PropTypes.func,
+  followType: PropTypes.string
+};
+
 const styles = StyleSheet.create({
-  buttonFollowing: {
+  buttonFollowing: (followType) => ({
     width: normalize(88),
     height: normalize(36),
     flexDirection: 'row',
     justifyContent: 'center',
     alignItems: 'center',
     paddingVertical: 6,
     borderWidth: 1,
-    borderColor: COLORS.signed_primary,
+    borderColor: followType === 'incognito' ? COLORS.anon_secondary : COLORS.signed_primary,
     borderRadius: 8
-  },
-  textButtonFollowing: {
+  }),
+  textButtonFollowing: (followType) => ({
     fontFamily: fonts.inter[600],
     fontWeight: 'bold',
     fontSize: normalizeFontSize(12),
-    color: COLORS.signed_primary
-  }
+    color: followType === 'incognito' ? COLORS.anon_secondary : COLORS.signed_primary
+  })
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The styles are now dynamically generated based on the `followType` prop. This is a good approach to handle different styles for different types of users. However, it might be more efficient to memoize these functions using `useMemo` to avoid unnecessary recalculations during re-renders.

```diff
 const styles = StyleSheet.create({
  buttonFollowing: (followType) => ({
     ...
  }),
  textButtonFollowing: (followType) => ({
     ...
  })
});
+
+ const memoizedStyles = useMemo(() => styles(followType), [followType]);
```
Then use `memoizedStyles` instead of `styles` inside your component.


<!-- This is an auto-generated comment by OSS CodeRabbit -->